### PR TITLE
[CBRD-24611] Typechecking without server interaction

### PR DESCRIPTION
**Pull request by hyunikn**
http://jira.cubrid.org/browse/CBRD-24611

Typechecking without server interaction and implicit type conversion

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,4 +1,3 @@
-
```
Comment (hgryoo) [inline .gitignore]: Please remove the unrelated changes

**Review by beyondykk9 — APPROVED** [review]

**Review by shparkcubrid — APPROVED** [review]

**Review by hgryoo — APPROVED** [review]

File: src/jsp/com/cubrid/plcsql/predefined/sp/SpLib.java [hunk 2]
```diff
@@ -263,7 +256,105 @@ public static Boolean opXor(Boolean l, Boolean r) {
 
     // ====================================
     // comparison equal
-    public static Boolean opEq(Object l, Object r) {
+    public static Boolean opEq(Boolean l, Boolean r) {
```
Comment (youngjinj) [inline src/jsp/com/cubrid/plcsql/predefined/sp/SpLib.java:259]: It's trivial.
What about reducing code with generic methods?

File: src/jsp/com/cubrid/plcsql/compiler/ast/TypeSpecSimple.java [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+    public static TypeSpecSimple INTEGER = of("java.lang.Integer");
+    public static TypeSpecSimple LONG = of("java.lang.Long");
+    public static TypeSpecSimple FLOAT = of("java.lang.Float");
+    public static TypeSpecSimple DOUBLE = of("java.lang.Double");
+    public static TypeSpecSimple LOCALDATE = of("java.time.LocalDate");
+    public static TypeSpecSimple LOCALTIME = of("java.time.LocalTime");
+    public static TypeSpecSimple ZONEDDATETIME = of("java.time.ZonedDateTime");
+    public static TypeSpecSimple LOCALDATETIME = of("java.time.LocalDateTime");
+    public static TypeSpecSimple SET = of("java.util.Set");
+    public static TypeSpecSimple MULTISET = of("org.apache.commons.collections4.MultiSet");
```
Comment (youngjinj) [inline src/jsp/com/cubrid/plcsql/compiler/ast/TypeSpecSimple.java:121]: Is it okay if only MultiSet uses the Apache Commons Collections API?
If it is decided to use the Apache Commons Collections API, I think it will be consistent to use SET and LIST together.

**Review by ctshim — APPROVED** [review]

**Review by youngjinj — APPROVED** [review]

**Comment by hyunikn** [comment]
.
